packages/kit/src/cli.js

```diff
@@ -85,10 +85,13 @@ prog
 	.option('-H, --https', 'Use self-signed HTTPS certificate')
 	.option('-o, --open', 'Open a browser tab')
 	.action(async ({ port, host, https, open }) => {
-		await check_port(port);
-
 		process.env.NODE_ENV = process.env.NODE_ENV || 'development';
 		const config = await get_config();
+		const vite_config = config.kit.vite();
+
+		if (!vite_config.server || vite_config.server.strictPort !== false) {
```

Comment:
As a work-around for the moment: if server is not set or strictPort is not false, do a check port